### DISCUSSION TITLE
findBy/findByOrFail with multiple criteria?

### DISCUSSION THREAD

**Discussion by Jeordy** [adonisjs/core#3750] [Ideas]
Hi all,

After testing Adonisjs for a while, I really love it. There's one small 'issue' when using the findBy and findByOrFail function on models:

I really would like to pass in an object with search criteria, in stead only passing one value only. 

`
const payload = {id: 2, tenantId: 1};

const user = await User.findOrFail(payload);
`
It's not possible to use the user.merge({...}) function when querying with the User.query(), because it's not using the model directly. 

Or maybe there's another way to query on multiple selectors and still use the .merge() function?

**Comment by WeyIin** [comment]
How about `User.query().where('id', 2).andWhere('tenantId', 1).firstOrFail()` ?

**Reply by Jeordy** [reply]
Thank you very much! I didn't use the .firstOrFail() and that triggered the .merge().

But I still think the findOrFail with a payload could be a great feature.

**Comment by andre-ab** [comment]
> Hi all,
> 
> After testing Adonisjs for a while, I really love it. There's one small 'issue' when using the findBy and findByOrFail function on models:
> 
> I really would like to pass in an object with search criteria, in stead only passing one value only.
> 
> ` const payload = {id: 2, tenantId: 1};
> 
> const user = await User.findOrFail(payload); ` It's not possible to use the user.merge({...}) function when querying with the User.query(), because it's not using the model directly.
> 
> Or maybe there's another way to query on multiple selectors and still use the .merge() function?

User this
```
// Assuming you have the Product model imported
import Product from 'App/Models/Product';

async function findProduct() {
  // Step 1: Define your variable with a dynamic column name and value
  const test = { column: 'id', value: 2 };

  // Step 2: Extract the column name and value from the variable
  const columnName = test.column;
  const columnValue = test.value;

  // Step 3: Pass the column name and value to findByOrFail
  try {
    const product = await Product.findByOrFail(columnName, columnValue);
    console.log(product);
  } catch (error) {
    console.error('Product not found', error);
  }
}
```
